### PR TITLE
nixos/mihomo: init

### PR DESCRIPTION
**Pull request by Guanran928**
## Description of changes

Adds a module for `clash-meta`, a rule-based proxy based on Go, similar to https://github.com/NixOS/nixpkgs/pull/250301.

I'm new to Nix/NixOS and nixpkgs, please tell if I did anything wrong. ^^

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/networking/clash-meta.nix [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    ### systemd service
+    systemd.services."clash-meta" = {
+      description = "Clash.Meta daemon, A rule-based proxy in Go.";
+      documentation = [ "https://wiki.metacubex.one/" ];
+      requires = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig =
+        {
+          ExecStart = builtins.concatStringsSep " " [
```
Comment (Aleksanaa) [inline nixos/modules/services/networking/clash-meta.nix:77]:
```suggestion
          ExecStart = lib.concatStringsSep " " [
```

For better compatibility, I guess?

**Review by Aleksanaa — COMMENTED** [review]
Very good work!

**Review by oluceps — APPROVED** [review]
LGTM

**Comment by doronbehar** [comment]
@ofborg test mihomo

**Comment by ThinkChaos** [comment]
Hi, I got here from the release-notes because I didn't find anything that looks like a "rule-based proxy in Go" at the linked repo: [MetaCubeX/mihomo](https://github.com/MetaCubeX/mihomo).  
The repo description is "A simple Python Pydantic model for Honkai: Star Rail parsed data from the Mihomo API."

I don't understand Chinese so maybe one of the other links in the PR clarifies this, but are the release notes correct?  
Can they be improved because the current situation is confusing.

**Comment by oluceps** [comment]
> Hi, I got here from the release-notes because I didn't find anything that looks like a "rule-based proxy in Go" at the linked repo: [MetaCubeX/mihomo](https://github.com/MetaCubeX/mihomo). The repo description is "A simple Python Pydantic model for Honkai: Star Rail parsed data from the Mihomo API."
> 
> I don't understand Chinese so maybe one of the other links in the PR clarifies this, but are the release notes correct? Can they be improved because the current situation is confusing.

About the naming and description issue please refer to https://github.com/NixOS/nixpkgs/pull/271658#discussion_r1518828185

**Comment by ThinkChaos** [comment]
Ok I found the actual project, it's in [a non default branch](https://github.com/MetaCubeX/mihomo/tree/meta-main). I think that that branch should be linked from the release notes instead of default branch which shows a project seemingly unrelated to the actual service exposed by the module.
Again maybe some info is lost in translation for me, but I suspect most people that read the release notes and check out the project will have the same confusion as I did.

**Comment by doronbehar** [comment]
Feel free to write a PR changing URLs or other links and ping me for a review.

**Pull request merged**